### PR TITLE
fix(webex-core): send metrics when service not found in service catalog

### PR DESCRIPTION
**Pull request by arun3528**
# COMPLETES https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-306551

## This pull request addresses

send metrics when there is no service url found in the catalog 

## by making the following changes

< DESCRIBE YOUR CHANGES >

In the services.js file if the url is. not found send a metrics to amplitude service you can check the url in this link 
https://analytics.amplitude.com/cisco-cross-project/space/qumao1s/all

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [x] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

Did manual testing 

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Comment by arun3528** [comment]
@InteractiveTimmy  @taymoork2   does the draft look good for you ?, moved the browser detection in common and some common metrics parameters in plugin metrics

File: packages/node_modules/@webex/webex-core/src/lib/services/services.js [hunk 1]
```diff
@@ -577,11 +578,15 @@ const Services = WebexPlugin.extend({
     }
 
     if (catalog.isReady) {
-      return url ?
-        Promise.resolve(url) :
-        Promise.reject(new Error(
-          `services: service '${name}' was not found in any of the catalogs`
-        ));
+      if (url) return Promise.resolve(url);
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/webex-core/src/lib/services/services.js:581]: Not sure if the ESLint config enforces this, but shouldn't this if statement be wrapped in `{` and `}`? JS does support this syntax, but wanting confirmation that ESLint does as well.

File: packages/node_modules/@webex/webex-core/src/lib/services/services.js [hunk 2]
```diff
@@ -577,11 +578,15 @@ const Services = WebexPlugin.extend({
...
+
+      this.webex.internal.metrics.submitClientMetrics(SERVICE_METRICS.JS_SDK_SERVICE_NOT_FOUND, {
+        fields: {service_name: name}
+      });
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/webex-core/src/lib/services/services.js:585]: Simple and clean!

File: packages/node_modules/@webex/common/src/browser-detection.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    getBrowserName,
+    getBrowserVersion,
+    isBrowser
+  };
+};
+
+export default memoize((agent) => (agent || window.navigator?.userAgent ?
+  createDetectionObject(bowser.getParser(agent || window.navigator.userAgent)) :
+  mockDetectionObject
+));
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/common/src/browser-detection.js:36]: This is a globally useful! Awesome work!

File: packages/node_modules/@webex/common/src/metrics.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+// Metrics for service catalog
+export const SERVICE_METRICS = {
+  JS_SDK_SERVICE_NOT_FOUND: 'JS_SDK_SERVICE_NOT_FOUND'
+};
+
+// Metric to do with WDM registration
+export const WDM_METRICS = {
+  JS_SDK_WDM_REGISTRATION_SUCCESSFULL: 'JS_SDK_WDM_REGISTRATION_SUCCESSFULL',
+  JS_SDK_WDM_REGISTRATION_FAILED: 'JS_SDK_WDM_REGISTRATION_FAILED'
+};
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/common/src/metrics.js:11]: This is amazing to see, This also fulfills a request i believe may exist in our backlog around getting usage/failure metrics from SDK explicit items.

**Review by InteractiveTimmy — APPROVED** [review]
A lot more than just a simple metrics addition here, but rather, an expansive/modular approach to amending better metrics! Great work!

Note: Testing may be needed, but I'll leave that jurisdiction up to the current code owners.

File: packages/node_modules/@webex/webex-core/src/lib/services/services.js (hunk 1 above)
Comment (arun3528) [reply packages/node_modules/@webex/webex-core/src/lib/services/services.js]: done

File: packages/node_modules/@webex/internal-plugin-metrics/src/metrics.js [hunk 5]
```diff
@@ -3,12 +3,42 @@
  */
 
 import {WebexPlugin} from '@webex/webex-core';
-import {deprecated} from '@webex/common';
 
+import BrowserDetection from '../../common/src/browser-detection';
```
Comment (taymoork2) [inline packages/node_modules/@webex/internal-plugin-metrics/src/metrics.js:7]: This should be imported using `@webex/common` not file pathed, this causes issues when bundling

File: packages/node_modules/@webex/webex-core/src/lib/services/services.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import Url from 'url';
 import sha256 from 'crypto-js/sha256';
 
 import WebexPlugin from '../webex-plugin';
+import {SERVICE_METRICS} from '../../../../common/src/metrics';
```
Comment (taymoork2) [inline packages/node_modules/@webex/webex-core/src/lib/services/services.js:6]: This should be imported via `@webex/common` and not file path'd

**Review by taymoork2 — CHANGES_REQUESTED** [review]

**Comment by arun3528** [comment]
@taymoork2  fixed the review comments

**Comment by arun3528** [comment]
@taymoork2  tested the netlify link here is the screen shot for the metrics

**Review by taymoork2 — APPROVED** [review]
LGTM

**Pull request merged**